Spigot-Server-Patches/0460-Add-basic-brand-support.patch

```diff
@@ -0,0 +1,60 @@
+From 887c56195ad8378f5a84eca12f681a1e58e74599 Mon Sep 17 00:00:00 2001
+From: DigitalRegent <misterwener@gmail.com>
+Date: Mon, 6 Apr 2020 17:14:57 +0200
+Subject: [PATCH] Add basic brand support
+
+
+diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
+index 7929fcc8..319f12f9 100644
+--- a/src/main/java/net/minecraft/server/PlayerConnection.java
++++ b/src/main/java/net/minecraft/server/PlayerConnection.java
+@@ -106,6 +106,8 @@ public class PlayerConnection implements PacketListenerPlayIn {
+     private static final int MAX_SIGN_LINE_LENGTH = Integer.getInteger("Paper.maxSignLength", 80);
+     private static final long KEEPALIVE_LIMIT = Long.getLong("paper.playerconnection.keepalive", 30) * 1000; // Paper - provide property to set keepalive limit
+ 
++    private String clientBrandName = null;
++
+     public PlayerConnection(MinecraftServer minecraftserver, NetworkManager networkmanager, EntityPlayer entityplayer) {
+         this.minecraftServer = minecraftserver;
+         this.networkManager = networkmanager;
+@@ -2664,6 +2666,10 @@ public class PlayerConnection implements PacketListenerPlayIn {
+                 PlayerConnection.LOGGER.error("Couldn\'t unregister custom payload", ex);
+                 this.disconnect("Invalid payload UNREGISTER!");
+             }
++        // paper start -- add brand support
++        }  else if(packetplayincustompayload.tag.equals(PacketPlayOutCustomPayload.a)) {
++            this.clientBrandName = packetplayincustompayload.data.readUTF(256);
```

Comment:
Should allow to set this only once? Thoughts?

Reply:
The brand name is not final for client 

[>  Additionally, the brand may change at any time](https://wiki.vg/Plugin_channels#minecraft:brand)

Reply:
Ah, TIL.

Reply:
I'd prefer if we didn't re-read the packet data here.

Reply:
They're never read before and/or after in any accessible scope.

Reply:
Alright then, nevermind.